wicker/core/persistance.py

```diff
@@ -225,30 +224,16 @@ def persist_wicker_dataset(
         # 4. Sort the dataset if not sorted
         sorted_dataset_0 = sorted(dataset_0, key=lambda tup: tup[0])
 
-        # 5. Partition the dataset into K partitions
-        partitions = []
-
-        def divide_chunks(list_to_divide):
-            # looping till length l
-            for i in range(0, len(list_to_divide), PARTITION_SIZE):
-                partitions.append(list_to_divide[i : i + PARTITION_SIZE])
-
-        divide_chunks(sorted_dataset_0)
```

Comment:
The partitions that the spark persistor is making are driven by the data (which comes in tuples [partition, row_data]) rather than the persistor chunking it.
In practice, our workflows will actually have data for one partition only ("train" or "eval") so there is a missed opportunity there. The train/eval should be separate datasets and the persistors should have flexibility to partition the dataset but that's for another day.